wgsl/index.bs

```diff
@@ -629,20 +631,28 @@ with all valid code points
 of both [=UAX31 Lexical Classes|XID_Start=] and
 [=UAX31 Lexical Classes|XID_Continue=].
 
-Note: A user agent or should issue developer-visible warnings when
-the meaning of a WGSL program would change if all instances of an
-identifier spelled with a specific code point sequence are replaced
-by another code point sequence that would appear the same to a reader.
-For example, remapping each identifier to its canonical equivalent
-spelling under [=UAX15 Normalization Forms|Normalization Form C (NFC)=]
-must permit the same set of declarations (avoiding new identifier collisions),
-and must not change how identifier uses [=resolve=] to their declarations.
-
 Note: The [=return type=] for some [=built-in functions=] are structure types whose name cannot be used WGSL source.
 Those structure types are described as if they were [=predeclared=] with a name starting with two underscores.
 The result value can be saved into newly declared `let` or `var` using type inferencing, or immediately have one of its members
 immediately extracted by name.  See example usages in the description of `frexp` and `modf`.
 
+### Identifier Comparison ### {#identifier-comparison}
+
+WGSL identifiers [=string/is|identical to=] each other are the same. As such, they do not compare
+assuming normalizations such as [=UAX15 Normalization Forms|Normalization Form C (NFC)=],
+[=UAX15 Normalization Forms|Normalization Form D (NFD)=], [=UAX15 Normalization Forms|Normalization Form KC (NFKC)=],
+[=UAX15 Normalization Forms|Normalization Form KD (NFKD)=],
+and further matching such as [=UTS46 Mapping=], and [=charmod-norm Matching=].
+
+Note: A user agent should issue developer-visible warnings when WGSL identifiers contain
```

Comment:
This is subtly different from the text it is replacing.  It's a strong statement in fact.

Consider this text:

         fn foo() { let  H = ... ; }
         fn bar() { let H = ...; }

Assume the two `H` have distinct code points but are homoglyphs for each other.  The scopes of those identifers don't overlap.   So the program doesn't change meaning if you replace instances of one with its homoglyph.


Suggest:

> Note:
A user agent should issue developer-visible warnings when the meaning of a WGSL program would change if all instances of an identifier are replaced with one of that identifier's homographs.
(A homoglyph is a sequence of code points that may appear the same to a reader as another sequence of code points. 
Examples of mappings to detect homoglyphs are the transformations, mappings, and matching algorithms mentioned in the previous paragraph.
Two sequences of code points are homographs if you can transform one into the other by repeatedly replacing a subsequence with its homoglyph.)

Reply:
@dneto0 Done, this is much better!